momentum/4/modules/ha-proxy-client.md

```diff
@@ -0,0 +1,58 @@
+---
+title: "ha_proxy_client - HAProxy protocol client module"
+description: "The ha_proxy_client module is used to configure Momentum to use HAProxy's PROXY protocol version 2 for outbound connections (see https://github.com/haproxy/haproxy/blob/master/doc/proxy-protocol.txt).  This can be leveraged in cases where your sending IPs are deployed on a different machine than the Momentum.
+---
+
+When configured for a binding or binding group Momentum will connect to the given `ha_proxy_server` and prefix the SMTP session with a PROXY protocol version 2 header.  The `dst_addr` and `dst_port` will be filled in with the resolved MX, the `src_addr` will be filled in with the configured value of `ha_proxy_src_addr`.  It is the customers responsibility to configure a listener on `dst_addr:dst_port` that listens for PROXY protocol and routes based on the `src_addr`, but a simple example configuration for HAProxy would be:
```

Comment:
The stuff on IPV6 could probably just be folded into here.  I think we could just say something like:

```
the `src_addr` will be filled in with the configured value of `ha_proxy_src_addr` if the destination family is IPV4, or `ha_proxy_ipv6_src_addr` if the destination family is IPV6.  If you need to deliver to both IPV4 and IPV6 destinations than you must configure both options for the binding or binding_group.
```